src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -108,14 +171,8 @@ internal bool CanCreateModel(MutableObjectBinderContext context)
                 return true;
             }
 
-            // 2. If it is top level object and there are no properties to bind
-            if (isTopLevelObject && context.PropertyMetadata != null && context.PropertyMetadata.Count == 0)
```

Comment:
Ah, got it. Previous condition was hidden -- have already checked `isTopLevelObject`.